CMakeLists.txt

```diff
@@ -263,145 +264,159 @@ endif ()
 
 set (FFTW_VERSION 3.3.7)
 
-set (PREC_SUFFIX)
-if (ENABLE_FLOAT)
-  set (FFTW_SINGLE TRUE)
-  set (BENCHFFT_SINGLE TRUE)
-  set (PREC_SUFFIX f)
+if (BUILD_SHARED_LIBS)
+  add_definitions (-DFFTW_DLL)
 endif ()
 
-if (ENABLE_LONG_DOUBLE)
-  set (FFTW_LDOUBLE TRUE)
-  set (BENCHFFT_LDOUBLE TRUE)
-  set (PREC_SUFFIX l)
-endif ()
+function (add_fftw_library PRECISION)
+  if (PRECISION MATCHES "SINGLE|LDOUBLE|QUAD")
+    set (FFTW_${PRECISION} TRUE)
+    set (BENCHFFT_${PRECISION} TRUE)
+    if (PRECISION STREQUAL SINGLE)
+      set (PREC_SUFFIX f)
+    else ()
+      string (SUBSTRING ${PRECISION} 0 1 PREC_SUFFIX)
+      string (TOLOWER ${PREC_SUFFIX} PREC_SUFFIX)
+    endif()
+  endif ()
 
-if (ENABLE_QUAD_PRECISION)
-  set (FFTW_QUAD TRUE)
-  set (BENCHFFT_QUAD TRUE)
-  set (PREC_SUFFIX q)
-endif ()
-set (fftw3_lib fftw3${PREC_SUFFIX})
+  set (fftw3_lib fftw3${PREC_SUFFIX})
 
-configure_file (cmake.config.h.in config.h @ONLY)
-include_directories (${CMAKE_CURRENT_BINARY_DIR})
+  add_library (${fftw3_lib} ${SOURCEFILES})
 
-if (BUILD_SHARED_LIBS)
-  add_definitions (-DFFTW_DLL)
-endif ()
+  configure_file (cmake.config.h.in ${CMAKE_CURRENT_BINARY_DIR}/${fftw3_lib}/config.h @ONLY)
+  target_include_directories (${fftw3_lib} PRIVATE "${CMAKE_CURRENT_BINARY_DIR}/${fftw3_lib}")
 
+  if (MSVC)
+    target_compile_definitions (${fftw3_lib} PRIVATE /bigobj)
+  endif ()
+  if (HAVE_SSE)
+    target_compile_options (${fftw3_lib} PRIVATE ${SSE_FLAG})
+  endif ()
+  if (HAVE_SSE2)
+    target_compile_options (${fftw3_lib} PRIVATE ${SSE2_FLAG})
+  endif ()
+  if (HAVE_AVX)
+    target_compile_options (${fftw3_lib} PRIVATE ${AVX_FLAG})
+  endif ()
+  if (HAVE_AVX2)
+    target_compile_options (${fftw3_lib} PRIVATE ${AVX2_FLAG})
+  endif ()
+  if (HAVE_FMA)
+    target_compile_options (${fftw3_lib} PRIVATE ${FMA_FLAG})
+  endif ()
+  if (HAVE_LIBM)
+    target_link_libraries (${fftw3_lib} m)
+  endif ()
 
-add_library (${fftw3_lib} ${SOURCEFILES})
-if (MSVC)
-  target_compile_definitions (${fftw3_lib} PRIVATE /bigobj)
-endif ()
-if (HAVE_SSE)
-  target_compile_options (${fftw3_lib} PRIVATE ${SSE_FLAG})
-endif ()
-if (HAVE_SSE2)
-  target_compile_options (${fftw3_lib} PRIVATE ${SSE2_FLAG})
-endif ()
-if (HAVE_AVX)
-  target_compile_options (${fftw3_lib} PRIVATE ${AVX_FLAG})
-endif ()
-if (HAVE_AVX2)
-  target_compile_options (${fftw3_lib} PRIVATE ${AVX2_FLAG})
-endif ()
-if (HAVE_FMA)
-  target_compile_options (${fftw3_lib} PRIVATE ${FMA_FLAG})
-endif ()
-if (HAVE_LIBM)
-  target_link_libraries (${fftw3_lib} m)
-endif ()
+  list (APPEND subtargets ${fftw3_lib})
 
-set (subtargets ${fftw3_lib})
+  if (ENABLE_THREADS AND Threads_FOUND)
+    if (WITH_COMBINED_THREADS)
+      target_link_libraries (${fftw3_lib} ${CMAKE_THREAD_LIBS_INIT})
```

Comment:
`Threads::Threads` available since CMake 3.1:
* https://cmake.org/cmake/help/v3.1/module/FindThreads.html

Use it intead of `CMAKE_THREAD_LIBS_INIT`.

Reply:
Same concerns as with `GenerateExportHeader`. Also, this will bump the minimum CMake version up to 3.1. I'm not sure if this is desired upstream. Ultimately, we'll want to be in sync with upstream so we can drop this fork, right? I can open a separate PR upstream though and see what they say. :)

Reply:
> Ultimately, we'll want to be in sync with upstream so we can drop this fork, right?

Also we want everything to work correctly. If it will not be accepted upstream, then just use fork.

Reply:
Understood. I'll change it o use `Threads::Threads` as we'll update to CMake 3.1 anyway because of `CMP0054`.